src/include/function/gds/gds.h

```diff
@@ -1,129 +1,80 @@
 #pragma once
 
-#include "binder/expression/expression.h"
+#include "common/mask.h"
+#include "function/table/bind_data.h"
+#include "graph/graph.h"
 #include "graph/graph_entry.h"
-#include "parser/query/reading_clause/yield_variable.h"
+#include "processor/result/factorized_table_pool.h"
 
 namespace kuzu {
-namespace binder {
-class Binder;
-}
+
 namespace main {
 class ClientContext;
 }
-namespace processor {
-class FactorizedTable;
-struct ExecutionContext;
-struct GDSCallSharedState;
-} // namespace processor
 
 namespace function {
 
-class PathLengths;
-
-struct GDSBindInput {
-    binder::expression_vector params;
-    binder::expression_vector optionalParams;
-    binder::Binder* binder = nullptr;
-    std::vector<parser::YieldVariable> yieldVariables;
-
-    GDSBindInput() = default;
-
-    std::shared_ptr<binder::Expression> getParam(common::idx_t idx) const { return params[idx]; }
-    std::shared_ptr<binder::Expression> getOptionalParam(common::idx_t idx) const {
-        return optionalParams[idx];
-    }
-    common::idx_t getNumParams() const { return params.size(); }
-};
-
-struct GDSBindData {
+struct GDSBindData : public TableFuncBindData {
     graph::GraphEntry graphEntry;
     std::shared_ptr<binder::Expression> nodeOutput;
 
-    GDSBindData(graph::GraphEntry graphEntry, std::shared_ptr<binder::Expression> nodeOutput)
-        : graphEntry{graphEntry.copy()}, nodeOutput{std::move(nodeOutput)} {}
+    GDSBindData(binder::expression_vector columns, graph::GraphEntry graphEntry,
+        std::shared_ptr<binder::Expression> nodeOutput)
+        : TableFuncBindData{std::move(columns)}, graphEntry{graphEntry.copy()},
```

Comment:
Is it possible for any gds algorithms to find out the numRows in the binding stage?
if possible, we'd better pass the `numRows` parameter to TableFuncBindData.

Reply:
I don't think GDS requires numRows to be known from upfront.